test/controllers/MessagesControllerSpec.scala

```diff
@@ -96,26 +103,36 @@ class MessagesControllerSpec extends AnyWordSpec with Matchers with GuiceOneAppP
   private val serviceConfig = new ServicesConfig(configuration)
   private val appConfig     = new AppConfig(configuration, serviceConfig)
 
-  private val controller = new MessagesController(appConfig, Helpers.stubControllerComponents())
+  private def controller(messageConnector: MessageConnector = mock[MessageConnector]) = new MessagesController(appConfig, Helpers.stubControllerComponents(), messageConnector)
```

Comment:
Why it was changed from `val` to `def` in that case? 🤔

Reply:
So that we can inject a message connector when we use it, but also provide a default mock when it doesn't have to do anything.